bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -156,6 +160,75 @@ private void refresh() {
+                            getSenecValue(response.grid.currentGridPowerPerPhase[0]).setScale(2, RoundingMode.HALF_UP),
+                            SmartHomeUnits.WATT));
+
+            Channel channelGridPowerPhase2 = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_GRID_POWER_PH2);
+            updateState(channelGridPowerPhase2.getUID(),
+                    new QuantityType<Power>(
+                            getSenecValue(response.grid.currentGridPowerPerPhase[1]).setScale(2, RoundingMode.HALF_UP),
+                            SmartHomeUnits.WATT));
+
+            Channel channelGridPowerPhase3 = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_GRID_POWER_PH3);
+            updateState(channelGridPowerPhase3.getUID(),
+                    new QuantityType<Power>(
+                            getSenecValue(response.grid.currentGridPowerPerPhase[2]).setScale(2, RoundingMode.HALF_UP),
+                            SmartHomeUnits.WATT));
+
+            Channel channelGridVoltagePhase1 = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_GRID_VOLTAGE_PH1);
+            updateState(channelGridVoltagePhase1.getUID(), new QuantityType<ElectricPotential>(
+                    getSenecValue(response.grid.currentGridVoltagePerPhase[0]).setScale(2, RoundingMode.HALF_UP),
+                    SmartHomeUnits.VOLT));
+
+            Channel channelGridVoltagePhase2 = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_GRID_VOLTAGE_PH2);
+            updateState(channelGridVoltagePhase2.getUID(), new QuantityType<ElectricPotential>(
+                    getSenecValue(response.grid.currentGridVoltagePerPhase[1]).setScale(2, RoundingMode.HALF_UP),
+                    SmartHomeUnits.VOLT));
+
+            Channel channelGridVoltagePhase3 = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_GRID_VOLTAGE_PH3);
+            updateState(channelGridVoltagePhase3.getUID(), new QuantityType<ElectricPotential>(
+                    getSenecValue(response.grid.currentGridVoltagePerPhase[2]).setScale(2, RoundingMode.HALF_UP),
+                    SmartHomeUnits.VOLT));
+
+            Channel channelGridFrequency = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_GRID_FREQUENCY);
+            updateState(channelGridFrequency.getUID(),
+                    new QuantityType<Frequency>(
+                            getSenecValue(response.grid.currentGridFrequency).setScale(2, RoundingMode.HALF_UP),
+                            SmartHomeUnits.HERTZ));
+
+            Channel channelBatteryStateValue = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_BATTERY_STATE_VALUE);
+            updateState(channelBatteryStateValue.getUID(),
+                    new DecimalType(getSenecValue(response.energy.batteryState).intValue()));
```

Comment:
You specified `Number:Dimensionless` so you should use a QuantityType here.

Reply:
Hello, but this is just a decimal without any Type. In the end it is a value of an enum. I don't find any good SmartHomeUnits Type for that. Do you have a proposal what I could give as second param if I go for `QuantityType<Dimensionless>`?

```
updateState(channelBatteryStateValue.getUID(),
    new QuantityType<Dimensionless>(getSenecValue(response.energy.batteryState).intValue()),
    SmartHomeUnits.**????**);
```

Reply:
SmartHomeUnits.ONE or Units.ONE

Reply:
Although if it is really an enum then you shouldn't specify it as `Number:Dimensionless` in the xml and instead just specify it as a `Number` instead.

Reply:
Done the change 👍